Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ubuntu:22.04
+
+RUN apt-get update && apt-get install -y \
+                    git \
+                    wget \
+                    flex \
+                    bison \
+                    gperf \
+                    python3 \
+                    python3-pip \
+                    python3-venv \
+                    python3-setuptools \
+                    cmake \
+                    ccache \
+                    libffi-dev \
+                    libssl-dev \
+                    dfu-util \
+                    libusb-1.0-0 \
+                    libgcrypt-dev \
+                    libglib2.0-dev \
+                    libfdt-dev \
+                    libpixman-1-dev \
+                    zlib1g-dev \
+                    ninja-build \
+                    libslirp-dev
+
+
+
+RUN mkdir -p ~/esp && cd ~/esp && git clone -b v4.4.7 --recursive https://github.com/espressif/esp-idf.git && cd esp-idf && ./install.sh && /bin/bash -c "source  ./export.sh"
```

Comment:
maybe it is worth adding `--depth 1` to the git clone, as it is taking quite significant amount of time to clone everything.